src/vmm/src/signal_handler.rs

```diff
@@ -193,31 +192,141 @@ mod tests {
         let child = thread::spawn(move || {
             assert!(register_signal_handlers().is_ok());
 
-            let filter = SeccompFilter::new(
-                vec![
-                    allow_syscall(libc::SYS_brk),
-                    allow_syscall(libc::SYS_exit),
-                    allow_syscall(libc::SYS_futex),
-                    allow_syscall(libc::SYS_getpid),
-                    allow_syscall(libc::SYS_munmap),
-                    allow_syscall(libc::SYS_kill),
-                    allow_syscall(libc::SYS_rt_sigprocmask),
-                    allow_syscall(libc::SYS_rt_sigreturn),
-                    allow_syscall(libc::SYS_sched_getaffinity),
-                    allow_syscall(libc::SYS_set_tid_address),
-                    allow_syscall(libc::SYS_sigaltstack),
-                    allow_syscall(libc::SYS_write),
-                ]
-                .into_iter()
-                .collect(),
-                SeccompAction::Trap,
-            )
-            .unwrap();
-
-            assert!(SeccompFilter::apply(filter.try_into().unwrap()).is_ok());
+            // Test with a seccomp filter that allows all syscalls, except for `SYS_mkdirat`.
+            // For some reason, directly calling `kick` with SIGSYS, like we do with the
```

Comment:
Not sure I get this comment. Do you mean directly calling `SYS_kill` with SIGSYS results in an error? SIGSYS is generated by `seccomp` or by issuing a syscall with bad arguments.